packages/messaging/src/ui/components/UnreadIcon/UnreadIcon.tsx

```diff
@@ -1,11 +1,223 @@
+/*
+    Copyright 2025 Adobe. All rights reserved.
+    This file is licensed to you under the Apache License, Version 2.0 (the
+    "License"); you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0 Unless required by applicable law
+    or agreed to in writing, software distributed under the License is
+    distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS OF
+    ANY KIND, either express or implied. See the License for the specific
+    language governing permissions and limitations under the License.
+*/
+import React, { useState } from 'react';
+import {
+  Image,
+  ImageProps,
+  ImageStyle,
+  StyleSheet,
+  View,
+  ViewProps,
+  ViewStyle,
+  useColorScheme
+} from 'react-native';
+import useContainerSettings from '../../hooks/useContainerSettings';
+
+export interface UnreadIconProps extends ViewProps {
+  imageStyle?: ImageStyle;
+  containerStyle?: ViewStyle;
+  source?: ImageProps['source'];
+  darkSource?: ImageProps['source'];
+  size?: number;
+  position?: 'top-left' | 'top-right' | 'bottom-left' | 'bottom-right';
+  type?: 'dot' | 'image';
+}
+
+// Helper function to convert placement from settings to component position
+const convertPlacement = (placement: 'topleft' | 'topright' | 'bottomleft' | 'bottomright'): 'top-left' | 'top-right' | 'bottom-left' | 'bottom-right' => {
+  switch (placement) {
+    case 'topleft':
+      return 'top-left';
+    case 'topright':
+      return 'top-right';
+    case 'bottomleft':
+      return 'bottom-left';
+    case 'bottomright':
+      return 'bottom-right';
+    default:
+      return 'top-right';
+  }
+};
+
+const UnreadIcon = ({
+  imageStyle,
+  containerStyle,
+  source,
+  darkSource,
+  size = 20,
+  position = 'top-right',
+  type = 'dot',
+  style,
+  ...props
+}: UnreadIconProps) => {
+  const colorScheme = useColorScheme();
+  const settings = useContainerSettings();
+  const [imageLoadError, setImageLoadError] = useState(false);
+  
+  // Get unread indicator settings from context
+  const unreadSettings = settings.content.unread_indicator;
+  
+  // Use settings from context with fallbacks to props
+  const displaySize = size;
+  const displayPosition = unreadSettings?.unread_icon?.placement ? 
+    convertPlacement(unreadSettings.unread_icon.placement) : position;
+  const renderType = unreadSettings?.unread_icon?.image ? 'image' : type;
+  const imageSource = unreadSettings?.unread_icon?.image?.url ? 
+    { uri: unreadSettings.unread_icon.image.url } : source;
+  const darkImageSource = unreadSettings?.unread_icon?.image?.darkUrl ? 
+    { uri: unreadSettings.unread_icon.image.darkUrl } : darkSource;
+  
+  const getPositionStyle = () => {
+    const baseStyle = {
+      position: 'absolute' as const,
+      zIndex: 1000,
+    };
+
+    switch (displayPosition) {
+      case 'top-left':
+        return { ...baseStyle, top: 6, left: 6 };
```

Comment:
since baseStyle is applied everywhere, you can just keep it in the styles of the component its applied to, Id also just add these styles in StyleSheet.create so they can be optimized